src/sql/src/session/vars.rs

```diff
@@ -3756,11 +3791,53 @@ impl SystemVars {
         *self.expect_value(&PERSIST_ROLLUP_THRESHOLD)
     }
 
-    pub fn persist_flags(&self) -> BTreeMap<String, bool> {
-        PersistFeatureFlag::ALL
-            .iter()
-            .map(|f| (f.name.to_owned(), *self.expect_value(&f.into())))
-            .collect()
+    pub fn persist_configs(&self) -> PersistConfigUpdates {
+        let mut updates = PersistConfigUpdates::default();
+        for entry in self.persist_configs.entries() {
+            // TODO: Well this is unfortuanate. The ConfigEntry has a shared
+            // value, but it's not actually connected to the corresponding
+            // `SystemVar`. We could fix that with another `SystemVarMut` impl,
+            // but that seems like overkill for initially landing the dyn_cfg
+            // stuff. Also, it's not clear if that's desirable. At the moment,
+            // the environmentd persist configs are only updated when the
+            // controllers hear about them and sharing the value would be a
+            // behavior change.
+            //
+            // Or we could just have two copies of ConfigSet in-mem in
+            // environmentd. One in adapter and one actually hooked up to the
+            // PersistClients that get created. That wouldn't be a behavior
+            // change but it might be confusing (i.e. the catalog has persist
+            // clients now, why do we have to update the controllers to get the
+            // config updates there?).
```

Comment:
FWIW I think this is basically fine, at least for now!

Reply:
Leaned into this and removed the HACK